File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        L, R = 1e9, 0
        
        for p in prices:
            L = min(L, p)
            R = max(R, p - L)
            
        return R
"""
1. len 10만, p[i] 1만
"""

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int min_price = 1e9;
        
        for(int price: prices) {
            min_price = min(min_price, price);
            profit = max(profit, price - min_price);
        }
        
        return profit;
    }
};